### PR TITLE
Ruby Capstone Project `Ruby` `PostgreSQL` `Rspec`

### PR DESCRIPTION
**Pull request by Ruthmy**
## Hi! @Luffytaro22, @alexmedinasf and @Ruthmy here. 👋

This is the PR for the [Ruby Capstone Project](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/catalog_of_my_things.md) of the Ruby curriculum at [Microverse](https://www.microverse.org/). The goal of this project is to build a command-line application that allows users to manage their collections of things. The application will allow users to create and read items in their collections.

---

## **Command-line Application Goals:**

The goals of this project are to:

* Create a command-line application that allows users to manage their collections of things that they own; books, music albums and games.
* Allow users to create and read items in their collections.
    * **List Items:** Users can retrieve a comprehensive list of all items in their collection, categorized by type (books, music albums, movies, and games). 
    * **Add Items:** Users have the capability to add new items to their collection. Whether it's a new book, music album or game, users can input details about their acquisitions.
    * **Archive Items:** The application offers an archiving feature that automatically determines if an item should be archived based on specific criteria (e.g., published date, condition, last played date).
* **Store the data in JSON files.** All data entered by users is stored persistently in JSON files, ensuring that their collections remain intact even after exiting the application.

---

## **Project Goals:**

- [x] Use a kanban board to manage project tasks and allocate work efficiently among team members. Ensure that all project requirements are appropriately distributed and completed.
- [x] Develop a database schema that mirrors the structure of the application's class hierarchy.
- [x] Follow the [GitFlow as a workflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) to create and unify the different features of the project.
- [x] Adhere to established [best practices for Ruby programming](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md), ensuring code quality, readability, and maintainability.
- [x] Documented the work in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md), including code comments and a video presentation. The video presentation should highlight the project's features and demonstrate its functionality.
- [x] Implement unit tests for all methods to verify the correctness of the codebase and promote code reliability.
- [x] Peer code reviews were performed for each of the delivered tasks, writing clear, professional, and respectful review comments for team members.
<br>
<h3><a href="https://drive.google.com/file/d/1op3tY8KSMCfPW9jXbZ60-WnTKDPjcycu/view?usp=sharing">Presentation Video - Ruby Capstone</a></h3>
<br>

### Thank you for your review ✨ 
<br>

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+CREATE TABLE Books(
+    id INT, 
+    name VARCHAR(255), 
+    published_date DATE, 
+    publisher VARCHAR(255), 
+    cover_state VARCHAR(255), 
+    archived BOOLEAN, 
+    author_id INT REFERENCES Author(id), 
+    label_id INT REFERENCES Label(id), 
+    genre_id INT REFERENCES Genre(id),
```
Comment (1GORDON) [inline schema.sql:77]: Great job team 🤝

- [ ] As stated in the project requirements, kindly ensure all foreign keys columns have corresponding indexes.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Ruthmy, @alexmedinasf, and @Luffytaro22 ✋

Good job so far!

## Highlights!
- Data is loaded from the JSON files on restart of the app ✔️
- Data is preserved in the JSON files ✔️
- App runs as expected ✔️
- All tests are passing ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Ruthmy) [reply schema.sql:77]: #### Hi @1GORDON 👋 
You are right, we will fix that.
Thank you for your feedback and your time in the review. ✨

**Review by Strangeal — APPROVED** [review]
Hi @Ruthmy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ruthmy** [comment]
Thank you very @Strangeal for your time and your review ✨ 
Happy weekend! 🥳

**Pull request merged**